301-multi-vmss-linux/nestedtemplates/base.json

```diff
@@ -80,11 +78,11 @@
           "storageProfile": {
             "osDisk": {
               "vhdContainers": [
-                "[concat('https://', variables('uniqueStringArray')[0], parameters('newStorageAccountPrefix'), '.blob.core.windows.net/', variables('vhdContainerName'))]",
-                "[concat('https://', variables('uniqueStringArray')[1], parameters('newStorageAccountPrefix'), '.blob.core.windows.net/', variables('vhdContainerName'))]",
-                "[concat('https://', variables('uniqueStringArray')[2], parameters('newStorageAccountPrefix'), '.blob.core.windows.net/', variables('vhdContainerName'))]",
-                "[concat('https://', variables('uniqueStringArray')[3], parameters('newStorageAccountPrefix'), '.blob.core.windows.net/', variables('vhdContainerName'))]",
-                "[concat('https://', variables('uniqueStringArray')[4], parameters('newStorageAccountPrefix'), '.blob.core.windows.net/', variables('vhdContainerName'))]"
+                "[concat(reference(resourceId('Microsoft.Storage/storageAccounts/', variables('uniqueStringArray')[0]), '2016-01-01').primaryEndpoints.blob, 'vhds')]",
```

Comment:
apiVersion isn't necessary in the reference() call if the resource is provisioned in the same template. The 'Microsoft.Storage/storageAccounts/' prefix should be necessary either.